src/components/Missions.js

```diff
@@ -1,7 +1,24 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { fetchMissions } from '../redux/missions/Missions';
 
-const Missions = () => (
-  <h1>Missions</h1>
+const Missions = () => {
+  const missionsData = useSelector((state) => state.missionsReducer);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchMissions());
+  }, []);
```

Comment:
Good job implementing actions to fetch the data and making it just run on the first render of the page, but we still can improove this code, every time you change view (like going from missions page to rocket page) it will dispatch the action for fetching data and is said on the requierments for this task to: Make sure you only dispatch those actions once and do not add data to store on every re-render (i.e. when changing views / using navigation).
Kindly add a condition so this actions doesn't get dispatch again on the same session
Hint: 
  useEffect(() => {
    if (missionsData.length === 0) dispatch(fetchMissions());
  }, [missionsData.length]);

Reply:
Great Walter.

Thanks for the lesson. I wasn't sure if I was coding right.